### DISCUSSION TITLE
Why do we have negative time in Audacity - do we need it?

### DISCUSSION THREAD

**Discussion by petersampsonaudacity** [audacity/audacity#4188] [Development]
I have always wondered why we have the concept of audio (and labels) that can be in negative time - and note carefully that this cannot be displayed at all on the screen unless the user turns on **Enable scrolling left of zero** in **Tracks Behaviors** preferences (and this setting is _**off**_ by default.

Does the concept of audio in negative time serve any useful purpose - I can't think of a single use-case?

A recent Forum Post: https://forum.audacityteam.org/viewtopic.php?f=46&t=128756&p=462526#p462526
led me to log the following three issues related to this:
- [ ]  #4185 
- [ ]  #4186 
- [ ]  #4187 
- [ ]  #4440

For my money I would advocate losing the concept of negative time in Audacity altogether _(although I note that Dominic Mazzoni enabled users to slide audio into negative time right from the get-go in 1.0.0 onwards)_.

Flagging: @Tantacrul @DilsonsPickles  and @LWinterberg  - as this is a design/UX discussion/decision

**Comment by LWinterberg** [comment]
I am strongly in favor of getting rid of negative time. It causes a bunch of problems, and the main benefit I can see (being able to non-destructively trim the beginning of an audio clip) got obsoleted with 3.1.

**Reply by LWinterberg** [reply]
Ran a quick [poll](https://www.youtube.com/post/UgkxbiknxSQWEqkEVEE9YvUapjKGHhwHLzaM) on it, and so far the only use case people reported was the above, with a ~2/3 to ~1/3 split in favor of "I don't use it". 

I see two potential solutions here: 

1. Remove negative time altogether, and accept that trimming now is a 2-step operation when it used to be one.
2. Remove negative time altogether but attempting to drag a clip before 0 automatically trims the beginning (after a bit of resistance).

**Comment by petersampsonaudacity** [comment]
@LWinterberg 

Your option 1 has the advantage that it wipes out all three of the issues listed in my initial post in this thread.

Option 2 means that those issues still have to be dealt with.

---------------------------------------------

I am fascinated by your poll Leo.  

Did those 1/3 of polled users tell you just what they used audio in negative time for?  Just what were the use cases?

Steve did provide one niche use case in the Forum thread that triggered this set of testing:
https://forum.audacityteam.org/viewtopic.php?f=46&t=128756&p=462468#p462468
For me that would not be sufficient cause to retain negative time.

**Reply by LWinterberg** [reply]
Count-in options need to be part of a proper metronome, they don't (or at least: shouldn't) require negative time. 

Regarding the bugs, they wouldn't be a issue: The selection would never move into negative time, instead it'd be trimmed. So if you have a 5s audio clip selected and moved it 1s to the left, the clip would have its first second "smart clip trimmed", and the selection would be shortened to 4s. 

That said: I feel like option 2 is very much overdesigning for a specific edge case which sort of clashes with the logic of the overlapping clips spec that's floating around somewhere.

**Comment by petersampsonaudacity** [comment]
Testing with Reaper I can't seem to be able to move audio into negative time - at least not with default settings, nor can I find any setting to make it so.

Nor can I move video into negative time with Microsoft's Video Editor on my PC.

**Reply by DavidBailes** [reply]
> Testing with Reaper I can't seem to be able to move audio into negative time - at least not with default settings, nor can I find any setting to make it so.

In Reaper, File > Project Settings.. , where you can change the project start time, and so have negative times.

**Reply by LWinterberg** [reply]
In practice Reaper's start time does not achieve the same as Audacity's:
* changing the start time moves all clips relative to your start time. So if you have a clip start at 2s and change the start time to 3s, the clip now starts at 5s. 
* Negative time is exported, with no indication that anything ever was in negative time
* you cannot move stuff before the timeline start

The use case for the Reaper (and other audio/video editor) behavior are imported files with SMPTE or MIDI timecodes and markers based on said timecodes. None of this stuff is in Audacity right now, and it's not going to be relevant for a very, very long time.

**Reply by DavidBailes** [reply]
> * changing the start time moves all clips relative to your start time. So if you have a clip start at 2s and change the start time to 3s, the clip now starts at 5s.

But if you set the start time to a negative number, then you get negative times. For example, starting with an item which starts at 12s:
![image](https://user-images.githubusercontent.com/11887021/212978716-30a92618-dd74-4114-a140-f267f96a0086.png)

Set project start time to -12s:
![image](https://user-images.githubusercontent.com/11887021/212978843-be9d2f13-a16f-47f1-af4c-97d402e0d417.png)

Use case given in Reaper manual:
![image](https://user-images.githubusercontent.com/11887021/212978324-47977328-ede7-4cbd-9dd7-b7dc8962ef75.png)

**Reply by LWinterberg** [reply]
Right. Which is different from what Audacity is doing

**Comment by petersampsonaudacity** [comment]
See also this recent Forum post re, negative time:

**Set timeline to start at 0 rather than negative number**
https://forum.audacityteam.org/viewtopic.php?f=47&t=129161
